### PR TITLE
Add scaling for detectors to SaveLauenorm

### LINKED ISSUE #17498: SaveLaueNorm have option to use detScale parameters
SaveLaueNorm should have a checkbox option to use the detScale parameters that are used in SaveHKL.  They can be set with SetDetScale algorithm for testing.  This is a request from MANDI instrument scientist.

### PR DESCRIPTION
**Pull request by VickieLynch**
Description of work.
SaveLaueNorm has a checkbox option to use the same instrument parameters that are used in SaveHKL for scaling detector banks. They can be set with SetDetScale algorithm for testing. This is a request from MANDI instrument scientist.  The unit test was modified to check that this works.
**To test:**
Check that the intensity and sigI are half after DetScale=0.5:
https://www.dropbox.com/s/kz3udbt2vbuouih/new.integrate?dl=0

``` python
import os

#load a peaks workspace from file
peaks = LoadIsawPeaks(Filename=r'new.integrate')
SetDetScale(Workspace='peaks', DetScaleList='1:0.5,2:1.5')
print "Number of peaks in table %d" % peaks.rowCount()

prefix = os.path.expanduser("~/MyPeaks")
for bool in [False, True]:
    SaveLauenorm(InputWorkspace=peaks, Filename=prefix, MinWavelength=1., MaxWavelength=2.,MinDSpacing=0.5, SortFilesBy='Bank',UseDetScale=bool)
    print "UseDetScale is set to ", bool
    finalfile = prefix + "001"
    ifile = open(finalfile, 'r')
    lines = ifile.readlines()
    for line in lines:
        print line
    ifile.close()
    print "Number of peaks in table %d" % len(lines)
```

Fixes #17498

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Crystal/test/SaveLauenormTest.h [hunk 1]
```diff
@@ -80,6 +80,42 @@ class SaveLauenormTest : public CxxTest::TestSuite {
...
+    TS_ASSERT_THROWS_NOTHING(alg2.setPropertyValue("Filename", outfile));
+    TS_ASSERT_THROWS_NOTHING(alg2.setProperty("UseDetScale", true));
+    TS_ASSERT_THROWS_NOTHING(alg2.execute(););
+    TS_ASSERT(alg2.isExecuted());
+    // Get the file
+    outfile = alg2.getPropertyValue("Filename") + "001";
+    fileExists = false;
+    TS_ASSERT(fileExists = Poco::File(outfile).exists());
+
+    if (fileExists) {
```
Comment (rosswhitfield) [inline Framework/Crystal/test/SaveLauenormTest.h]: Isn't this always `True` because you assert it above?

**Comment by rosswhitfield** [comment]
In your example if I compare bank 2 I expect it to increase 50%, as you set `2:1.5`, right? But I see the peak intensities are unchanged.

**Comment by VickieLynch** [comment]
bank 2 is scaled correctly.  The problem is that the banks are sorted 1,11,12,13,...  so that bank2 is in file 008.  I will fix the sort.

**Comment by VickieLynch** [comment]
I have emailed Leighton to see if changing the sort will make comparison with old data difficult.

**Comment by VickieLynch** [comment]
Leighton responded that the sort does not need to be changed.  He did request an option to not write out peaks from  1 or more detectors.

**Comment by rosswhitfield** [comment]
LGTM :shipit:

**Pull request merged**